src/storage/src/hummock/sstable/block_iterator.rs

```diff
@@ -252,7 +252,11 @@ impl BlockIterator {
                     let prefix =
                         self.decode_prefix_at(probe as usize, key_len_type, value_len_type);
                     let probe_key = &self.block.data()[prefix.diff_key_range()];
-                    match KeyComparator::compare_encoded_full_key(probe_key, key) {
+                    let mut buf: BytesMut = BytesMut::default();
+                    buf.put_u32(self.block.table_id());
+                    buf.put_slice(probe_key);
+                    let full_probe_key = FullKey::decode(&buf[..]);
```

Comment:
You don't need to create a buf to decode `&[8]` into `FullKey`.

We can add the following method in FullKey
```rust
impl<'a> FullKey<&'a [u8]> {
   ...
   /// Construct a [`FullKey`] from a byte slice without table_id encoded.
    pub fn from_slice_without_table_id(
        table_id: TableId,
        slice_without_table_id: &'a [u8],
    ) -> Self {
        let epoch_pos = slice_without_table_id.len() - EPOCH_LEN;
        let epoch = (&slice_without_table_id[epoch_pos..]).get_u64();

        Self {
            user_key: UserKey::new(table_id, TableKey(&slice_without_table_id[..epoch_pos])),
            epoch,
        }
    }
   ...
}
```

and use this method to decode slice without table_id encoded into FullKey here and in L80